design/KR/kenward.Rmd

```diff
@@ -0,0 +1,402 @@
+
+df_1d(result, contrast = c(0, 1), df = "KR1")
+# provide what we have in current `df_1d` results
+```
+
+## Rationals
+
+Why we implement the Kenward-Roger df in `mmrm` instead of using the `pbkrtest` package directly?
+
+* Current `mmrm` implementation is kind of complete, with the tests already implemented. If we are going to rely on `pbkrtest`
+package to implement Kenward-Roger degree of freedom, we are not able to include this method in `mmrm` otherwise we can introduce
+cyclic imports
+* `pbkrtest` is not well tested and we need effort to reliablely use that package
+* `pbkrtest` implements the original Kenward-Roger degree of freedom instead of the improved one(recommended)
+* SAS software provides both Kenward-Roger degree of freedom
+* `lmerTestR` suggests that "The culprit is the calculation of the scaling of the $F$-value for which **pbkrtest** does not appear to export a low-level (direct) method."
+which leads to slow speed which we can improve
+
+# Prototypes
+
+## mathmatical details for KR1
+
+### adjusted variance-covariance matrix
+
+Following the vignettes' notations, we have
+\[
+  Y \sim N(X\beta, \Sigma)
+\]
+where $\Sigma$ is a block diagonal matrix.
+Let
+\[
+  \Phi(\sigma) = {\left\{X^\top \Sigma(\sigma)^\top X\right\}}^{-1}
+\]
+\[
+  \hat\beta = \Phi(\hat\sigma) X^\top \Sigma(\hat\sigma)^{-1} Y
+\]
+
+$\Phi$ is the variance-covariance matrix for $\hat\beta$, and we usually use
+$\hat\Phi = \Phi(\hat\sigma)$ to measure the precision of $\hat\beta$.
+
+However, Kackar and Harville shows that the variability can be further partitioned into
+\[
+  V(\hat\beta) = \Phi + \Lambda
+\]
+where $\Lambda$ represents the asymptotic variance-covariance matrix underestimates $V(\hat\beta)$,
+and $\Lambda$ can be approximated by
+\[
+  \Lambda \simeq \Phi \left\{\sum_{i=1}^r{\sum_{j=1}^r{W_{ij}(Q_{ij} - P_i \Phi P_j)} }\right\} \Phi
+\]
+where
+\[
+  P_i = X^\top \frac{\partial{\Sigma^{-1}}}{\partial \sigma_i} X
+\]
+\[
+  Q_{ij} = X^\top \frac{\partial{\Sigma^{-1}}}{\partial \sigma_i} \Sigma \frac{\partial{\Sigma^{-1}}}{\partial \sigma_j} X
+\]
+and $W_{ij}$ is the $(i, j)$ element of $V(\hat\sigma)$.
+
+Further consider the bias of $\hat\Phi$ as an estimator of $\Phi$ using Taylor expansion
+
+\[
+  \hat\Phi \simeq \Phi + \sum_{i=1}^r{(\hat\sigma_i - \sigma_i)\frac{\partial{\Phi}}{\partial{\sigma_i}}} + \frac{1}{2} \sum_{i=1}^r{\sum_{j=1}^r{(\hat\sigma_i - \sigma_i)(\hat\sigma_j - \sigma_j)\frac{\partial^2{\Phi}}{\partial{\sigma_i}\partial{\sigma_j}}}}
+\]
+
+Ignoring the bias and the expectation is
+\[
+  E(\hat\Phi) = \Phi + \frac{1}{2} \sum_{i=1}^r{\sum_{j=1}^r{W_{ij}\frac{\partial^2{\Phi}}{\partial{\sigma_i}\partial{\sigma_j}}}}
+\]
+
+Using previous notations,
+
+\[
+  \frac{\partial^2{\Phi}}{\partial{\sigma_i}\partial{\sigma_j}} = \Phi (P_i \Phi P_j + P_j \Phi P_i - Q_{ij} - Q_{ji} + R_{ij}) \Phi
+\]
+where
+\[
+  R_{ij} = X^\top\Sigma^{-1}\frac{\partial^2\Sigma}{\partial{\sigma_i}\partial{\sigma_j}} \Sigma^{-1} X
+\]
```

Comment:
so I think this part is not needed to repeat from the paper, unless we have specific parts where we want to adapt the notation to what we use in the model fitting vignette

Reply:
This can be moved to the vignettes part